python/ray/data/read_api.py

```diff
@@ -1266,6 +1269,11 @@ def read_tfrecords(
         tf_schema=tf_schema,
     )
 
+    if schema_inference and not tf_schema:
```

Comment:
@scottjlee one thing that this PR doesn't cover, when the fallback read is applied, the schema is already infered and we should not try to infer it again. 
We could check here whether tfx-bsl is available as an extra condition, but seems like repeating things 

The other approach that I can think of is to not do the schema inference in the slow read, and always do it here. It is pretty fast to do it this way since the data should be already in memory (we did some benchmarks on our side and it indeed didn't add much processing time on some of our datasets)

Thoughts?

Reply:
@scottjlee my point was to avoid this:
```
Slow path succeeds, tf_schema not specified: currently does unwrap [[code](https://github.com/ray-project/ray/pull/34629/files#diff-9d628d90039025fb559945855019cefac89439c7f1a588bf2ba95b47656eb242R272-R276)]
```

because it leads to a pretty big overhead in performance, and instead use the unwrapping logic provided in this PR

Reply:
If i understand correctly, don't we need `tfx-bsl` to do the schema inference as you describe above? And the slow path is there as a backup for users who do not have `tfx-bsl`, so this may not be compatible?

Reply:
yeah, that would work, let me refactor it this way

Reply:
@scottjlee I think giving the option to disable the schema inference is nice, but I will let you guys decide if you agree or not, if you feel strongly I can delete.

regarding the unwrapping in `_get_feature_value `, I am not very familair with that code, so I was wondering if you will be able to do that either in this PR or in another?